server/application-server/src/main/java/de/tum/cit/aet/helios/environment/EnvironmentService.java

```
@@ -253,6 +253,95 @@ private OffsetDateTime getLockReservationExpiresAt(Environment environment) {
     }
   }
 
+  /**
+   * Extends the lock duration for an already locked environment.
+   *
+   * <p>This method attempts to extend the lock duration by resetting the lockWillExpireAt and
+   * lockReservationExpiresAt timestamps based on the current time. The environment must already be
+   * locked by the current user.
+   *
+   * <p>This method is transactional and handles optimistic locking failures.
+   *
+   * @param id the ID of the environment to extend the lock for
+   * @return an Optional containing the updated environment if successful, or an empty Optional if
+   *     the environment isn't locked, is locked by another user, or if an optimistic locking
+   *     failure occurs
+   * @throws EntityNotFoundException if no environment is found with the specified ID
+   * @throws IllegalStateException if the environment is disabled or isn't a TEST environment
+   */
+  @Transactional
+  public Optional<Environment> extendEnvironmentLock(Long id) {
+    final User currentUser = authService.getUserFromGithubId();
+
+    Environment environment = environmentRepository
+        .findById(id)
+        .orElseThrow(() -> new EntityNotFoundException("Environment not found with ID: " + id));
+
+    // Check environment status
+    if (!environment.isEnabled()) {
+      throw new IllegalStateException("Environment is disabled");
+    }
+
+    if (environment.getType() != Environment.Type.TEST) {
+      throw new IllegalStateException("Only TEST environments can have their locks extended");
+    }
+
+    // Verify the environment is locked and by the current user
+    if (!environment.isLocked() || !currentUser.equals(environment.getLockedBy())) {
+      return Optional.empty();
```

We can also throw the error here instead of in the controller.